lib/css_parser/parser.rb

```diff
@@ -263,7 +263,7 @@ def parse_block_into_rule_sets!(block, options = {}) # :nodoc:
       current_media_query = ''
       current_declarations = ''
 
-      block.scan(/([\\]?[{}\s"]|(.[^\s"{}\\]*))/).each do |matches|
```

Comment:
We run into an issue here where css can be malformed or malicious. Assume the following test:

```css
evil-div-one:before {
  content: "\\";
} 
evil-div-two:before {
  content: "\"";
}
```
If we use the current regex, it will have the following as tokens:
*evil-div-one tokens:*
"
\
\"

*evil-div-two tokens*
"
\"
"

For the second div, this is what we want, since it is quote character we want escaped. But in the first div, the last token (\") should actually be the real quote, not an escaped quote, as to turn off "in_string". The current result means that  a whole bunch of other css elements are pulled in as the token (since we never turned off in_string) until we hit another string.